src/builders/libraryBuilder.ts

```diff
@@ -89,9 +89,16 @@ export class LibraryBuilder {
 
       logger.debug(`generating CQL definitions for profile ${structureDef.name}`);
 
-      const resourceHandler: Handler | null = createHandler(structureDef, this.valueSets);
-      if (resourceHandler !== null) {
-        resources.push(resourceHandler.process());
+      if (structureDef.type in handlerLookup) {
+        const resourceHandlerClass : typeof Handler = handlerLookup[structureDef.type];
+        const resourceHandler: Handler | null  = new resourceHandlerClass(structureDef, this.valueSets);
```

Comment:
I believe I can remove the `| null` part with the way it is set up now, because `resourceHandlerClass` has to be of type `Handler`.